src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs

```diff
@@ -0,0 +1,522 @@
+                        copy.EnableStreaming = true;
+                        copy.DestinationTableName = targetTable;
+                        copy.WriteToServer(reader);
+                        copy.Close();
+                    }
+                }
+            }
+        }
+
+        internal void VerifyTablesEqual(string sourceTable, string targetTable, string connectionString)
+        {
+            using (SqlConnection connSource = new SqlConnection(connectionString))
+            {
+                connSource.Open();
+
+                using (SqlConnection connTarget = new SqlConnection(connectionString))
+                {
+                    connTarget.Open();
+
+                    using (SqlCommand cmdSource = new SqlCommand($"SELECT [c1], [c2] FROM [{tableNames["TabIntSourceDirect"]}] ORDER BY c1", connSource))
+                    using (SqlCommand cmdTarget = new SqlCommand($"SELECT [c1], [c2] FROM [{tableNames["TabIntSourceDirect"]}] ORDER BY c1", connTarget))
+                    {
+                        using (SqlDataReader sourceReader = cmdSource.ExecuteReader())
+                        using (SqlDataReader targetReader = cmdTarget.ExecuteReader())
+                        {
+                            while (sourceReader.Read() && targetReader.Read())
+                            {
+                                Assert.True(sourceReader.GetInt32(0) == targetReader.GetInt32(0), "Index mismatch");
+                                Assert.True(sourceReader.GetInt32(1) == targetReader.GetInt32(1), "Row mismatch");
+                            }
+                            sourceReader.Close();
+                            targetReader.Close();
+                        }
+                    }
+                }
+            }
+        }
+
+        internal void SilentRunCommand(string commandText, SqlConnection connection)
+        {
+            // Execute the command and swallow all exceptions
+            try
+            {
+                ExecuteQuery(connection, commandText);
+            }
+            catch (SqlException)
+            {
+                // Swallow!
+            }
+        }
+
+        internal void PopulateTable(string tableName, string valueS, string connectionString)
+        {
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+                ExecuteQuery(connection, $@"INSERT INTO [{tableNames[tableName]}] values ({valueS})");
+            }
+        }
+
+        public void Dispose()
+        {
+            foreach (string connection in DataTestUtility.AEConnStringsSetup)
+            {
+                using (SqlConnection sqlConnection = new SqlConnection(connection))
+                {
+                    sqlConnection.Open();
+                    foreach (string val in tableNames.Values)
+                    {
+                        Table.DeleteData(val, sqlConnection);
```

Comment:
Why do we need to delete the data on Dispose? Doesn't the fixture just drop all these tables when it is disposed?

Reply:
It was a duplication of IDisposable in couple of AE tests. Removed from test classes. SqlSetupStrategy already implements IDisposable.